### PR TITLE
Add nested numismatic place to numismatic issue

### LINKED ISSUE #2741: Add nested place form to numismatic issue

### PR DESCRIPTION
**Pull request by eliotjordan**
- Creates a numismatic place resource
- Adds nested numismatic place to numismatic issues
- Ensures that all nested resources are populated during changeset syncing.
- Creates a common populator `populate_nested_properties` in ChangeSet for all single-valued nested properties to use.
- Creates a common `prepopulate!` method in ChangeSet so that it doesn't have to be re-implemented in each changeset with a nested resource.

Closed #2741

### CODE REVIEW

File: app/change_sets/change_set.rb [hunk 1]
```diff
@@ -76,4 +76,48 @@ def fix_multivalued_keys
...
+      send(populator, fragment: fields, as: name)
+    end
+  end
+
+  # Override prepopulate method to correctly populate nested properties.
+  def prepopulate!(_args = {})
+    schema.each(twin: true) do |property|
+      property_name = property[:name]
+      property_klass = model.class.schema[property_name.to_sym]
+      send(:"#{property_name}=", property_klass.new) unless send(property_name)
```
Comment (tpendragon) [inline app/change_sets/change_set.rb]: If `twin: true` finds only the nested ones for some reason (like the above comment), can you link to that documentation?

File: app/change_sets/change_set.rb [hunk 2]
```diff
@@ -76,4 +76,48 @@ def fix_multivalued_keys
...
+    end
+
+    super
+  end
+
+  # Trigger population of nested properties when syncing changeset.
+  # Overrides {Disposable::Twin#sync}
+  def sync
```
Comment (tpendragon) [inline app/change_sets/change_set.rb]: Why is this necessary if you're adding the populator to the necessary properties in the ChangeSet?

File: app/change_sets/change_set.rb (hunk 1 above)
Comment (eliotjordan) [reply app/change_sets/change_set.rb]: I found this in reform. It uses it to find nested properties to validate: https://github.com/trailblazer/reform/blob/master/lib/reform/contract/validate.rb#L44-L47

File: app/change_sets/change_set.rb (hunk 2 above)
Comment (eliotjordan) [reply app/change_sets/change_set.rb]: The populator is not run when syncing the resource outside of a controller action. Instantiating a resource with the nested property in FactoryBot is an example. I'm sure why exactly, but I guess this is something buried within reform. Without the populator, when you run `sync` on the changeset, an exception is raised because reform (?) has mapped that nested property to a changeset.

Comment (eliotjordan) [reply app/change_sets/change_set.rb]: @tpendragon I created an branch with an example that raises the same exception with the existing letter change set. 

https://github.com/pulibrary/figgy/compare/nested-error-example

Comment (tpendragon) [reply app/change_sets/change_set.rb]: Does that still error if for sender you instead set it as just the properties and not the object?
